include/linux/kfence.h

```diff
@@ -59,13 +59,22 @@ void __init kfence_init(void);
  * kfence_shutdown_cache() - handle shutdown_cache() for KFENCE objects
  * @s: cache being shut down
  *
- * Return: true on success, false if any leftover objects persist.
- *
  * Before shutting down a cache, one must ensure there are no remaining objects
- * allocated from it. KFENCE objects are not referenced from the cache, so
- * kfence_shutdown_cache() takes care of them.
+ * allocated from it. Because KFENCE objects are not referenced from the cache
+ * directly, we need to check them here.
+ *
+ * Note that shutdown_cache() is internal to SL*B, and kmem_cache_destroy() does
+ * not return if allocated objects still exist: it prints an error message and
+ * simply aborts destruction of a cache, leaking memory.
```

Comment:
I thought we could maybe just call kfence_shutdown_cache() after __kmem_cache_shutdown(), so that we mimic the default kernel behavior, but this approach is better.